frontends/web/src/hooks/api.ts

```diff
@@ -60,6 +59,33 @@ export const useLoad = <T>(
                 }
             });
         }, // we pass no dependencies because it's only queried once
-        []); // eslint-disable-line react-hooks/exhaustive-deps
+        [] // eslint-disable-line react-hooks/exhaustive-deps
+    );
+    return respose;
+}
+
+/**
+ * useSync is a hook to load a promise and sync to a subscription function.
+ * It is a combination of useLoad and useSubscribe.
+ * gets fired on first render, and returns undefined while loading,
+ * re-renders on every update.
+ */
+export const useSync = <T>(
+    apiCall: () => Promise<T>,
```

Comment:
the two args could be replaced with `endpoint: string`, so the caller doesn't have to provide two functions that always look the same. or is there a benefit in defining the two functions in mobiledata.ts?

Reply:
`mobiledata.ts` is not the best example as it just returns a boolean as arg or in a promise. 

For me the main motivation of `src/api` is that this is the interface to the backend and the place where all types from the api are defined. Before they were defined in various components.

I think `endpoint: string` would be nice, but I am not sure where the types should live. Maybe a giant endpoint/returntypes map somewhere?